Demobot dancebot (#64)
ut-ras/ras_site: _posts/2020-05-03-dancebot.md

* We can use assets that represent the state of charge for the robots - a yawning robot face if the battery voltage dips to a low operational level, and a snoring robot face if the battery voltage dips to a critically low operational level. 
* An auto off circuit if the state of charge becomes too low. This will grey out the robot in the webpage after the heartbeat requests stop for at least a certain period of time. Perhaps something like [this](http://www.zolalab.com.br/eletronica_projetos/auto_shutdown_enus.php)?
* Image processing to automatically adjust the state of the robots and the Mothership, I.E. making the swarm do a certain dance or the Mothership expression change based on a pose classifier. Part of this work has already been done for the SIMLab at UT, so it's possible to integrate this with the Mothership's onboard RPI camera.

I would like for most, if not all of the design work to be done by the end of the summer. In terms of software development, I think that the webpage is a pretty easy project and can be done with a relatively low amount of effort weekly (1-3 hours for 12 weeks. This can probably be done in about 20 man hours max.).

> For long term plans, I would like to begin the construction, implementation, and testing of the robots this fall. I would love we could have one or two Version 2 Dancebots ready by the end of the semester, being able to be controlled by an RPI standalone. Having a working prototype of the Mothership complete as well will be a pretty big stretch goal.

Author: ```suggestion
For long term plans, I would like to begin the construction, implementation, and testing of the robots this fall. I would love if we could have one or two Version 2 Dancebots ready by the end of the semester, with the ability to be controlled by a RPI standalone. Having a working prototype of the Mothership complete as well will be a pretty big stretch goal.
```
